woocommerce/woocommerce-ios
# [Analytics Hub] Refactor Time Range data structure

Part of #8146 

#### Mostly repositioning and renaming class names and groups, don't be scared of the number of lines changed. 😅  

Why
==========
The previous `AnalyticsHubTimeRangeGenerator` class would be hard to maintain as we need to introduce new range selections like `last month`, `last year`, and new types like Quarter ranges. We need a simple way to define those ranges as they have a pattern repetition, configuring those ranges for reuse but keeping the current API.

How
==========
This PR introduces dedicated time range classes responsible for building the main ranges based on a reference date. The reference date can be changed to support different selection types, allowing those classes to be reused to generate types like `last month`, `yesterday, `last week` and `last year`. 

The `AnalyticsHubViewModel` will continue to only refer to the `AnalyticsHubTimeRangeSelection`, and that class has a delegation system that will call incorporate the correct range class based on the selected range type.

How to Test
==========
1. Launch the app and click the `See More button in the My Store charts section.
2. Verify that the date ranges and range selections are working as expected.

Update release notes:
